pkg/asset/installconfig/aws/validation.go

```diff
@@ -471,53 +485,35 @@ func validateDuplicateSubnetZones(fldPath *field.Path, subnets Subnets, idxMap m
 
 func validateServiceEndpoints(fldPath *field.Path, region string, services []awstypes.ServiceEndpoint) field.ErrorList {
 	allErrs := field.ErrorList{}
-	ec2Endpoint := ""
+	endpointOverrides := make(map[string]string)
 	for id, service := range services {
-		err := validateEndpointAccessibility(service.URL)
-		if err != nil {
-			allErrs = append(allErrs, field.Invalid(fldPath.Index(id).Child("url"), service.URL, err.Error()))
-			continue
-		}
-		if service.Name == ec2.ServiceName {
-			ec2Endpoint = service.URL
-		}
-	}
-
-	if partition, partitionFound := endpoints.PartitionForRegion(endpoints.DefaultPartitions(), region); partitionFound {
-		if _, ok := partition.Regions()[region]; !ok && ec2Endpoint == "" {
-			err := validateRegion(region)
+		if !sets.New(requiredServices...).Has(service.Name) {
+			err := validateEndpointAccessibility(service.URL)
 			if err != nil {
-				allErrs = append(allErrs, field.Invalid(fldPath.Child("region"), region, err.Error()))
+				allErrs = append(allErrs, field.Invalid(fldPath.Index(id).Child("url"), service.URL, err.Error()))
 			}
 		}
-		return allErrs
+		endpointOverrides[service.Name] = service.URL
 	}
 
-	resolver := newAWSResolver(region, services)
 	var errs []error
+	// validate the service endpoint for each of the required services. This will be the default endpoint for
+	// services where the user did not override the service endpoint through the install-config.
 	for _, service := range requiredServices {
-		_, err := resolver.EndpointFor(service, region, endpoints.StrictMatchingOption)
-		if err != nil {
-			errs = append(errs, fmt.Errorf("failed to find endpoint for service %q: %w", service, err))
+		if urlOverride, ok := endpointOverrides[service]; ok {
+			if err := validateEndpointAccessibility(urlOverride); err != nil {
```

Comment:
It seems like we're validating the endpoints twice... once here and once on line 491?